### ISSUE TITLE
Use `snforge_std` from package registry in `snforge init` template 

### ISSUE THREAD

**Issue by cptartur** [foundry-rs/starknet-foundry#2587]
### Description

Template used by `snforge init` uses `snforge_std` from github instead of https://scarbs.xyz/. 
Change it so it uses the latter.

### Context
- [Documentation about registries](https://docs.swmansion.com/scarb/docs/reference/specifying-dependencies.html#specifying-dependencies-from-registries)
- [Link to current template](https://github.com/foundry-rs/starknet-foundry/tree/bc608465bc72b1c84648bbae6558182deccd9e35/starknet_forge_template)

**Comment by Gerson2102** [comment]
Hey can I work on this one?

**Comment by ptisserand** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/starknet-foundry2).

### My background and how it can be leveraged
I'm a senior cairo developer with strong experience in Starknet ecosystem (since cairo 0). I have contributed to several projects on Only Dust writing smart contract code and rust code.
Also since I'm using starknet-foundry for my projects it will be really nice to have 

### How I plan on tackling this issue
I will update the template and then ensure that generated content of Scarb.toml will be ok.
I will also look to replace this manual check by an automatic one.

**Comment by akintewe** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/starknet-foundry).

### My background and how it can be leveraged
I'm experienced with package management systems and testing frameworks, having developed in JavaScript, Cairo, and Rust. I've dealt with a variety of package registries and see the need of standardizing dependency sources.


### How I plan on tackling this issue
I would approach this template adjustment by first looking at the present snforge init template structure, which relies on GitHub dependencies. I would change the template to utilize the scarbs.xyz registry for snforge_std, which ensures proper versioning and compatibility.
I would make sure the template is integrated correctly and with no errors. :)

**Comment by DanielEmmanuel1** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/starknet-foundry).

### My background and how it can be leveraged
Good day, my name is Deon, a Full -Stack Web3 Developer with a passion for building efficient and reliable software. 

I have experience in various Rust ecosystems and enjoy tackling challenges that improve code quality and user experience. My background includes working on open-source projects, contributing to Rust community discussions, OD Hack included and continuously learning about the latest advancements in Rust.

### How I plan on tackling this issue
-  I will start by cloning the repository where the snforge init template is defined.

- I'll locate the section of the code responsible for setting up the template and referencing snforge_std.

- Then, I'll change the reference from the GitHub link to https://scarbs.xyz/.

- After that is done, I'll run snforge init to ensure the changes work as expected and that no issues arise from the new template then update any relevant documentation to reflect every of the changes i made.

- Once that is done I'll submit a pull request with the changes, including a clear description of the modifications made.

I’m looking forward to resolving this issue and can't wait to get started on it! If given the opportunity i know i can deliever on this issue. Do have a great day.

**Comment by khayss** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/starknet-foundry).

### My background and how it can be leveraged
I’m a Rust developer with experience in both blockchain and open-source projects. I’ve contributed to various codebases where I focused on keeping dependencies up-to-date and optimizing code workflows. I’m well-versed in Rust’s package management.

### How I plan on tackling this issue
To update the `snforge init` template to use `snforge_std` from the package registry at `https://scarbs.xyz/`, I will:

1. Locate the current configuration in the `snforge init` template that points to `snforge_std` from GitHub.
2. Update the template to reference `snforge_std` from the correct package registry (`https://scarbs.xyz/`).
3. Test the initialization process to ensure the template works correctly with the new source.
4. Verify that all dependencies are resolved as expected and make sure the package functions seamlessly after the change.

I’d love to be considered for this issue.

---

Feel free to adjust as needed!

**Comment by akintewe** [comment]
created a PR @cptartur  Please review
